cabal-install/tests/IntegrationTests2.hs

```diff
@@ -1938,13 +1940,17 @@ testNixFlags = do
   Nothing @=? (fromFlag . globalNix . fromJust $ nixDefaultFlags)
 
   -- Config file options
-  defaultConfig <- loadConfig verbosity (Flag (basedir </> "nix-config/default-config"))
+  defaultConfig <- createDefaultConfigFile verbosity [] (basedir </> "nix-config/default-config")
   trueConfig <- loadConfig verbosity (Flag (basedir </> "nix-config/nix-true"))
   falseConfig <- loadConfig verbosity (Flag (basedir </> "nix-config/nix-false"))
 
   Nothing @=? (fromFlag . globalNix . savedGlobalFlags $ defaultConfig)
   Just True @=? (fromFlag . globalNix . savedGlobalFlags $ trueConfig)
   Just False @=? (fromFlag . globalNix . savedGlobalFlags $ falseConfig)
+
+  defaultConfigFile <- readFile (basedir </> "nix-config/default-config")
+  True @=? isInfixOf "-- nix:\n" defaultConfigFile
+
```

Comment:
can we generalize this test to ensure that essentially all options are commented out in the default config, not just nix?

Reply:
i.e. the repo perhaps and a few other things should not be commented, but it would be much better to have a much more uniform test.